From b3df0d88a24c156fb0a5a8a4699240e621841f47 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:06 -0400
Subject: [ethernet] i40e: Setting i40e_down bit for tx_timeout

Message-id: <1409658042-22689-6-git-send-email-sassmann@redhat.com>
Patchwork-id: 90201
O-Subject: [RHEL7.1 PATCH 005/281] i40e: Setting i40e_down bit for tx_timeout
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Akeem G Abodunrin <akeem.g.abodunrin@intel.com>

If tx_timeout recovery failed, then it becomes necessary to set
i40e_down bit before actually shutdown the connection.

Change-ID: Iaac81df0e302116571827aa0cff450697fbb7fa3
Signed-off-by: Akeem G Abodunrin <akeem.g.abodunrin@intel.com>
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit e108b0e341ec835df19dd3433b883b0abc509ed8)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index f596f74..8e44411 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -305,6 +305,7 @@ static void i40e_tx_timeout(struct net_device *netdev)
   break;
  default:
   netdev_err(netdev, "tx_timeout recovery unsuccessful\n");
+  set_bit(__I40E_DOWN, &vsi->state);
   i40e_down(vsi);
   break;
  }
-- 
1.7.1